## Task 4: Attributes of the Registration Table

Identify the data types of all attributes in `Zulassungen_alle_Fahrzeuge.csv` and describe their characteristics.

In [5]:
import pandas as pd

# Import the relevant file
registration = pd.read_csv("data/Zulassungen/Zulassungen_alle_Fahrzeuge.csv", sep=";",)

display(registration.head())
display(registration.info())

,Unnamed: 0,IDNummer,Gemeinden,Zulassung
0,408097,11-1-11-1,DRESDEN,2009-01-01
1,408098,11-1-11-2,DRESDEN,2009-01-01
2,1,12-1-12-1,LEIPZIG,2009-01-01
3,2,12-1-12-2,LEIPZIG,2009-01-01
4,3,12-1-12-3,DORTMUND,2009-01-01


<class 'pandas.DataFrame'>
RangeIndex: 3204104 entries, 0 to 3204103
Data columns (total 4 columns):
 #   Column      Dtype
---  ------      -----
 0   Unnamed: 0  int64
 1   IDNummer    str  
 2   Gemeinden   str  
 3   Zulassung   str  
dtypes: int64(1), str(3)
memory usage: 200.5 MB


None

The registration table `Zulassungen_alle_Fahrzeuge.csv` contains 3,204,104 entries with 4 columns and no missing values.

| Attribute | Pandas dtype | Actual Data Type | Scale Level | Description |
|---|---|---|---|---|
| Unnamed: 0 | int64 | Integer (index) | Discrete | Row index from CSV export, not a meaningful attribute |
| IDNummer | str | String (categorical) | Nominal | Unique vehicle identifier, e.g. `11-1-11-1` |
| Gemeinden | str | String (categorical) | Nominal | Municipality name, e.g. `DRESDEN` |
| Zulassung | str | Date (temporal) | Discrete | Registration date, stored as string, e.g. `2009-01-01` |

**Data Type Characteristics:**

- **Nominal:** Classification and labeling. Values can only be compared for equality (A = B or A ≠ B). No ranking or ordering possible. Examples: `IDNummer`, `Gemeinden`.
- **Ordinal:** Ordered values with ranking (A > B > C), but differences between values are not meaningful. Not present in this table.
- **Discrete:** Finite value range. Values can be counted and compared. `Unnamed: 0` is a row counter, `Zulassung` contains individual calendar dates (finite set of distinct values).
- **Continuous:** Infinite value range with meaningful differences. Not present in this table.

In [6]:
# Convert registration date from string to datetime
registration["Zulassung"] = pd.to_datetime(registration["Zulassung"])
print(registration.dtypes)

Unnamed: 0             int64
IDNummer                 str
Gemeinden                str
Zulassung     datetime64[us]
dtype: object
